src/fixes.rs

```diff
@@ -0,0 +1,147 @@
+use crate::common::*;
+
+mod lowercase_key;
+
+trait Fix {
+    fn name(&self) -> &str;
+
+    fn fix_warnings(
+        &self,
+        warnings: Vec<&mut Warning>,
+        lines: &mut Vec<LineEntry>,
+    ) -> Option<usize> {
+        let mut count: usize = 0;
+        for warning in warnings {
+            let line = &mut lines[warning.line_number() - 1];
+            if self.fix_line(line).is_some() {
+                warning.set_fixed(true);
+                count += 1;
+            }
+        }
+
+        Some(count)
+    }
+
+    fn fix_line(&self, _line: &mut LineEntry) -> Option<()> {
+        None
+    }
+
+    fn set_fixed_on_all(&self, warnings: Vec<&mut Warning>) -> Option<usize> {
+        let count = warnings.len();
+        for warning in warnings {
+            warning.set_fixed(true);
+        }
+
+        Some(count)
+    }
+}
+
+// TODO: skip fixes (like checks)
+// The fix order is matter
+fn fixlist() -> Vec<Box<dyn Fix>> {
+    vec![
+        // At first we run the fixers that handle a single line entry (they use default
+        // implementation of the fix_warnings() function)
+        Box::new(lowercase_key::LowercaseKeyFixer::default()),
+        // Then we should run the fixers that handle the line entry collection at whole.
+        // And at the end we should run the fixer for ExtraBlankLine check (because the previous
+        // fixers can create additional extra blank lines).
+    ]
+}
+
+pub fn run(warnings: &mut [Warning], lines: &mut Vec<LineEntry>) -> usize {
+    if warnings.is_empty() {
+        return 0;
+    }
+
+    for warning in warnings.iter_mut() {
+        warning.set_fixed(false);
+    }
+
+    let mut count = 0;
+    for fixer in fixlist() {
+        // We can optimize it: create check_name:warnings map in advance
```

Comment:
It would be very good to optimize this, the `group_by` function + storing the `fixlist` in the form of a `HashMap<String, Box<dyn Fix>>` would help a lot.

We can use lightweight [itertools](https://crates.io/crates/itertools) for this, because it really will allow to distribute all the `Warnings` in their checker names in 1 iteration, but [it has specifics](https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.into_group_map):

```
let warning_groups = warnings
   .iter_mut()
   .map(|w| (w.name(), w))
   .into_group_map();

for (k, v) in warnings_map.inter() {
   let fixer = fixers_map.get(k);
   // ... fix warnings 
}
```

But I think someone else needs an opinion on this :thinking: 
@dotenv-linter/core, what do you think?

Reply:
I can do this optimization now, or we can leave this for a separate issue.
I think there is no need for additional libraries.

Reply:
Let's leave it for the future. Maybe, someone else will do it.
And I wouldn't want to add an additional dependency.